### PR TITLE
LPS-111943 Historical data error handling

### PR DESCRIPTION
**Pull request by dgarciasarai**
Review here https://github.com/wincent/liferay-portal/pull/220

### CODE REVIEW

**Comment by dgarciasarai** [comment]
New commits since the review:

* 5f198f7 LPS-111943 Rename file to ConnectionContext
* 1eb7263 LPS-111943 Else/if: conditions are exclusive
* https://github.com/wincent/liferay-portal/pull/222/commits/3b7d2cb40ffc609f662dacd3f0325aafd0e10089 LPS-111943 Simplify conditions
* 0fa7fde LPS-111943 Add addWarning to the list of dependencies in useEffect
* https://github.com/wincent/liferay-portal/pull/222/commits/bf73ed750ec0a7253f7bbd3057805ad595f13319 LPS-111943 Simplify custom hook useWarning
* 60af27b LPS-111943 Simplify using immutable style to reduce the cognitive load
* 9fdcd0f LPS-111943 Fix doc-style comments

**Comment by wincent** [comment]
ci:test:sf

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function TotalCount({
 
 	const isMounted = useIsMounted();
 
-	const addWarning = useAddWarning();
+	const [, addWarning] = useWarning();
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js:36]: Nice. ❤️ destructuring assigment.

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/store.js [hunk 2]
```diff
@@ -34,14 +34,18 @@ export function StoreContextProvider({children}) {
...
 			type: ADD_WARNING,
 		});
 	}, [dispatch]);
+
+	const hasWarning = state.warning;
+
+	return [hasWarning, addWarning];
 }
 
 export function useHasWarning() {
```
Comment (wincent) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/store.js:51]: Let's delete this as I don't think it is used or needed any more.

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js (hunk 1 above)
Comment (dgarciasarai) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/TotalCount.js:36]: ❤️

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/store.js (hunk 2 above)
Comment (dgarciasarai) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/context/store.js:51]: Ok, I forgot to change it in the `AnalyticsReportsApp` file. I'm going to amend the change with this commit:
* bf73ed7 LPS-111943 Simplify custom hook useWarning

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
ci:stop

**Comment by dgarciasarai** [comment]
I'm going to resent this PR because React tests were failing.

**Pull request closed without merging**